Make ui_next the default UI

fosterseth opened the pull request:
### How to test
- download nodejs 18 https://nodejs.org/en/download/package-manager
- checkout branch
- locally (not in container), run `make clean-ui ui-devel`
- start development environment with `make docker-compose`
- navigate to `https://localhost:8043/` and login. Should see new UI
- install legacy UI by following these instructions https://github.com/ansible/awx/blob/ui_next_default/tools/docker-compose/README.md#install-legacy-ui
- make sure legacy UI works

### make target changes
New:
`ui-legacy`
`ui-legacy-release`
`clean-ui-legacy`

Changes (run these OUTSIDE awx containers, requires nodejs >= 18)
`ui-devel` - this now makes new UI
`clean-ui` - cleans new UI files

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Make the newer UI next the default UI experience.

Serves up old UI at `/ui_legacy`

pairs with https://github.com/ansible/awx-operator/pull/1923

![image](https://github.com/user-attachments/assets/235eb50b-3567-4179-9f0b-504d46b80b7e)

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.

-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New or Enhanced Feature

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - API

##### TODO

- [ ] remove welcome popup from ansible-ui
- [x] remove "tech preview" language from banner
- [x] make banner point to `/ui_legacy/` instead of `/`

fosterseth committed:
Make ui_next default

Make the newer UI next the default UI
experience.

Enable legacy ui

https://<controller_url>/ui_legacy

will redirect to the old user interface

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

gundalow commented on awx/ui/src/screens/Dashboard/Dashboard.js:
```diff
@@ -79,8 +79,8 @@ function Dashboard() {
             <p>
-              <InfoCircleIcon /> A tech preview of the new {brandName} user
-              interface can be found <a href="/ui_next">here</a>.
+              <InfoCircleIcon /> The new {brandName} user
+              interface can be found <a href="/">here</a>.
```
I might be miss understanding this, though it reads to me as "Go to `/` to enable the new UI", which should already be the default.

fosterseth replied:
these files (`awx/ui`) are the old ui files. So if the user is navigating the old UI, the banner will point them back to new UI (`/`)

samccann commented:
Just a general comment - having links with 'here' as the text don't work well with screen-readers. It is better to have something more descriptive in that bit of text.

fosterseth commented on awx/main/views.py:
```diff
@@ -61,38 +61,6 @@ class APIException(exceptions.APIException):
         return render(request, 'error.html', kwargs, status=status)
 
 
-def handle_400(request, exception):
```
dead code

fosterseth committed:
remove unused code

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

fosterseth committed:
return 404 if not enabled

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

fosterseth committed:
prettier

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

gundalow commented on awx/settings/defaults.py:
```diff
@@ -1090,6 +1090,7 @@
 CLUSTER_HOST_ID = socket.gethostname()
 
 UI_NEXT = True
+UI_LEGACY = False
```
Why do we have two variables, rather than one variable which is set to the UI type (constant)

fosterseth replied:
yeah this change is meant to be additive in nature. There are handful of ui places that depend on UI_NEXT setting so remove this setting will require a bunch of other changes.

we have some followup work outlined that will clean this up though.

fosterseth replied:
it wasn't a lot of work to remove UI_NEXT setting, so I went ahead and remove it

gundalow commented on awx/ui/src/screens/Dashboard/Dashboard.js:
```diff
@@ -79,8 +77,8 @@ function Dashboard() {
             <p>
-              <InfoCircleIcon /> A tech preview of the new {brandName} user
-              interface can be found <a href="/ui_next">here</a>.
+              <InfoCircleIcon /> The new user interface can be found{' '}
```
Please don't use "here"
Maybe you the link should be "enable new user interface"

fosterseth replied:
reworded it, see photo in opening body

gundalow commented on awx/ui/src/screens/SubscriptionUsage/SubscriptionUsage.js:
```diff
@@ -21,16 +20,15 @@ const MainPageSection = styled(PageSection)`
             <p>
-              <InfoCircleIcon /> A tech preview of the new {brandName} user
-              interface can be found <a href="/ui_next">here</a>.
+              <InfoCircleIcon /> The new user interface can be found{' '}
```
As above

AlanCoding commented:
![Screenshot from 2024-07-19 14-49-20](https://github.com/user-attachments/assets/d93c2b3a-baf7-4b0a-abd7-58ba0bd9cded)

When I go to the dev environment I get this. What do I need to run the dev environment with the new UI?

AlanCoding commented:
In the container I get:

```
bash-5.1$ make ui-next
make: Nothing to be done for 'ui-next'.
```

gundalow commented:
> ![Screenshot from 2024-07-19 14-49-20](https://private-user-images.githubusercontent.com/1385596/350523751-d93c2b3a-baf7-4b0a-abd7-58ba0bd9cded.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE0MTU0MDgsIm5iZiI6MTcyMTQxNTEwOCwicGF0aCI6Ii8xMzg1NTk2LzM1MDUyMzc1MS1kOTNjMmIzYS1iYWY3LTRiMGEtYWJkNy01OGJhMGJkOWNkZWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDcxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA3MTlUMTg1MTQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTMyMWZhN2E5ZmQ3ZjM1OGMxODhjYTZlZTFkOWFmMzgyNDliOThkMTQwMWMxYmRjZjA2ZWY0ZDYxOWY3NzdmZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6-dQ1kUcMxHkYdaoxjjJVtrEvzOJI_xogYXRlv2N-S0)
> 
> When I go to the dev environment I get this. What do I need to run the dev environment with the new UI?

Do we need a check in the makefile to validate the UI is available, ie fail early.

AlanCoding commented:
@gundalow I just want to know how to use this the first place. I read `awx/ui_next/README.md` (from hints in root Makefile) and set `UI_NEXT_GIT_REPO` to the git address for https://github.com/ansible/ansible-ui and the target still gives the same result.

fosterseth commented:
to use
1. from awx repo run `make ui-next` (outside the container)
2. make docker-compose

> Do we need a check in the makefile to validate the UI is available, ie fail early.
@gundalow 

this isn't too different than how it was before, where users had to manually run `make ui-devel` after starting containers

fosterseth committed:
remove UI_NEXT setting

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

fosterseth committed:
lint

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

fosterseth committed:
linter

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

gundalow commented:
> this isn't too different than how it was before, where users had to manually run `make ui-devel` after starting containers

Does this need to go into the contributing/devel docs somewhere?

fosterseth committed:
Remove UI_NEXT setting in UI src

fosterseth committed:
bump nodejs

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

fosterseth committed:
touch up make commands

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

fosterseth commented:
> Does this need to go into the contributing/devel docs somewhere?
@gundalow 

I added some documentation on installing the legacy UI

djyasin reviewed (approved):
I was able to pull down these changes and walk through the steps provided to successfully build ui_next and then the legacy UI with no issues.

TheRealHaoLiu committed:
multi-stage build

instead of building the ui during make sdist build the ui ahead of time in different docker build stage

fosterseth committed:
build ui outside of container

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

fosterseth committed:
add ui legacy to misc edit

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

fosterseth committed:
add ui_legacy entries in settings

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

fosterseth committed:
remove node_version

Signed-off-by: Seth Foster <fosterbseth@gmail.com>

fosterseth commented:
no longer plan to serve old UI alongside new UI

closing in favor of https://github.com/ansible/awx/pull/15405
